### DISCUSSION TITLE
REST: Status Codes

### DISCUSSION THREAD

**Discussion by davewhit3** [gofr-dev/gofr#677] [Q&A]
**Describe the bug**
I don't have option to set response code. [[source code]](https://github.com/gofr-dev/gofr/blob/development/pkg/gofr/http/responder.go#L53-L62)
In REST with a POST method and a status code of `201` I could send a location header, but it could be `200`, `202` or `204`.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201

The code is

   ```go
	if err == nil {
		switch r.method {
		case http.MethodPost:
			return http.StatusCreated, nil
		case http.MethodDelete:
			return http.StatusNoContent, nil
		default:
			return http.StatusOK, nil
		}
	}
   ```

**Expected behavior**
I can set the response code and headers.

**Environments (please complete the following information):**
- OS: Linux
- gofr version v1.6.1
- go version 1.22

**Comment by vipul-rawat** [comment]
Hey @davewhit3 we are trying to limit the user to not deviate from the REST standards - so 201 response for created resource. We will support 202 and 204 for ACCEPTED and NO CONTENT. But we would not agree on sending 200 for a POST request.

**Comment by davewhit3** [comment]
@vipul-rawat ok, but what if you don't create any new resource? 
What about access to headers?

**Comment by vipul-rawat** [comment]
@davewhit3 so if there is no creation of data, the POST method should not be used according to RESTful design.

As for response headers we are in talks about how to support headers while keeping it simple for users to not worry about them too much.

**Comment by davewhit3** [comment]
@vipul-rawat 

> @davewhit3 so if there is no creation of data, the POST method should not be used according to RESTful design.

It's not that clear. 
The POST method might not result in a resource that can be identified by a URI.
Sometimes we need to return an object with HTTP 200 and sometimes it should be HTTP 201 with a Location header.
Please think about it with your team.

> As for response headers we are in talks about how to support headers while keeping it simple for users to not worry about them too much.

Great! I'm waiting for it!
